lib/broadway_kafka/brod_client.ex

```diff
@@ -143,7 +143,7 @@ defmodule BroadwayKafka.BrodClient do
         {:ok, _} ->
           current_offset
 
-        {:error, :offset_out_of_range} ->
+        {:error, _} ->
```

Comment:
```suggestion
        # We are looking for :offset_out_of_range but this may fail
        # in earlier Kafka versions, so we perform a catch-all.
        {:error, _} ->
```